coderd/database/dbtestutil/db.go

```diff
@@ -19,9 +23,30 @@ func WillUsePostgres() bool {
 	return os.Getenv("DB") != ""
 }
 
-func NewDB(t testing.TB) (database.Store, pubsub.Pubsub) {
+type options struct {
+	fixedTimezone string
+}
+
+type Option func(*options)
+
+// WithTimezone sets the database to the defined timezone instead of
+// to a random one.
+//
+// Deprecated: If you need to use this, you may have a timezone-related bug.
```

Comment:
if this is going to be deprecated, then maybe a ruleguard?

Reply:
Honestly, it may make sense to remove the deprecation as there may be reasons to test with multiple database timezones.

But a lint rule for `WithTimeZone("UTC")` is a good idea.

Reply:
Linter rule added.